test/ci_workflow_gen.go

```diff
@@ -213,12 +220,44 @@ type clusterTest struct {
 }
 
 type vitessTesterTest struct {
-	FileName         string
-	Name             string
-	RunsOn           string
-	GoPrivate        string
-	GoJunitReportSHA string
-	Path             string
+	*GitMetas
+	FileName  string
+	Name      string
+	RunsOn    string
+	GoPrivate string
+	Path      string
+}
+
+// getGitMetas concurrently fetches Git metadata for workflow dependencies.
+func getGitMetas(ctx context.Context) (*GitMetas, error) {
+	var metasMu sync.Mutex
+	var metas GitMetas
+
+	eg, egCtx := errgroup.WithContext(ctx)
+
+	// vitessio/go-junit-report (uses HEAD)
+	eg.Go(func() error {
```

Comment:
I don't understand why we're using errgroups and locks here. Why would we not just make these two calls synchronously when we generate the workflows? It's fine, just seems like overkill to me.